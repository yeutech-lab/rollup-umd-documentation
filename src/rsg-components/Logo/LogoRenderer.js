import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

const styles = ({ color, fontFamily, fontSize }) => ({
	logo: {
		color: color.base,
		margin: 0,
		fontFamily: fontFamily.base,
		fontSize: fontSize.h4,
		fontWeight: 'normal',
	},
});

export function LogoRenderer({ classes, children }) {
	return <h1 className={classes.logo}>{children}</h1>;
}

LogoRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node,
};

export default Styled(styles)(LogoRenderer);
