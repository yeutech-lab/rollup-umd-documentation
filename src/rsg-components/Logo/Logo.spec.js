import React from 'react';
import LogoRenderer from './LogoRenderer';

it('renderer should render header', () => {
	const actual = render(<LogoRenderer>React Styleguidist</LogoRenderer>);

	expect(actual).toMatchSnapshot();
});
