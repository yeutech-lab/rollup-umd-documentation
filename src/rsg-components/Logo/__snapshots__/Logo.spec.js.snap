// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render header 1`] = `
<h1
  class="rsg--logo-0"
>
  React Styleguidist
</h1>
`;
